src/advanced/soft-play-advanced.js

```diff
@@ -0,0 +1,174 @@
+// do not change these lines
+// let adults = 0
+// let children = 0
+// let adultsLeaving = 0
+// let childrenLeaving = 0
+
+// TODO: Write your functions in the below section. Your functions should update
+// the adults and children variables defined above.
+// Start with the occupancy function.
+class Softplay {
+  constructor(entranceRatio, occupancyRatio) {
+    this.entranceRatio = entranceRatio // Add as adult:child
+    this.occupancyRatio = occupancyRatio // Add as adult:child
+    this.adults = 0
+    this.children = 0
+    this.adultsLeaving = 0
+    this.childrenLeaving = 0
+    this.occupancyValueTotal = {
+      adults: 0,
+      children: 0
+    }
+  }
+}
+// Example object to show the configuration for other softplays to be added to the platform
+
+// softplay = {
+//   entranceRatio: '1:1',
+//   occupancyRatio: '1:1',
+//   adults: 0,
+//   children: 0,
+//   adultsLeaving: 0,
+//   childrenLeaving: 0,
+//   occupancyValueTotal: {
+//     adults: 0,
+//     children: 0
+//   }
+// }
+
+const softplay = new Softplay('1:1', '1:1')
+
+const softplay2 = new Softplay('1:1', '1:1')
+
+const softplay3 = new Softplay('1:1', '1:1')
+
+function occupancy(softPlayCentre) {
+  const occupancyValue = {}
+  occupancyValue.adults = softPlayCentre.adults
+  occupancyValue.children = softPlayCentre.children
+
+  if (occupancyValue.adults === undefined || occupancyValue.adults === '') {
+    occupancyValue.adults = 0
+  }
+
+  if (occupancyValue.children === undefined || occupancyValue.children === '') {
+    occupancyValue.children = 0
+  }
+
+  return occupancyValue
+}
+function enter(numAdults, numChildren, softPlayCentre) {
+  entranceRatio = softPlayCentre.entranceRatio.split(':')
+
+  console.log(Number(entranceRatio[0]) + ' ' + Number(entranceRatio[1]))
+  if (Number(entranceRatio[0]) === 0 || Number(entranceRatio[1]) === 0) {
+    throw new Error(
+      'Your softplay entrance ratio is not configured correctly, please ensure that neither value is 0'
```

Comment:
You're using this string a couple of times, I'd move it to a const so if it ever needs to change you only need to update it in one place. You could create a `constants.js` or `errorMessages.js` file to import them from for use